File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    int left = 0;
    int right = 0;
    
    int sum = left;
    while(right <= n)
    {
        if (sum <= n)
        {
            if (sum == n)
                ++answer;
            
            if (right >= n)
                break;
            
            sum += ++right;
        }
        else if (sum > n)
        {
            sum -= left;
            ++left;
        }
    }
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for (int i = 1; i <= n; i++) {
            int num = i;
            int sum = 0;
            while(sum < n) {
                sum += num;
                num++;
            }
            if (sum == n) {
                answer++;
            }
        }
        return answer;
    }
}